### PR TITLE
CORE-5913: Change params from Role and User HTTP RPC endpoints

### PR DESCRIPTION
**Pull request by vkolomeyko**
Mostly switching from body params to path parameters.

### CODE REVIEW

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/role/RoleEndpoint.kt [hunk 1]
```diff
@@ -47,22 +47,22 @@ interface RoleEndpoint : RpcOps {
     /**
      * Associates a role with a permission
      */
-    @HttpRpcPUT(path = "addPermission", description = "Add a permission to a role")
+    @HttpRpcPUT(path = "addPermission/{roleId}/{permissionId}", description = "Add a permission to a role")
```
Comment (sprintengine) [inline libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/role/RoleEndpoint.kt]: Matthew [linked](https://r3holdco.slack.com/archives/C039BDESELW/p1659100130181029?thread_ts=1659097839.695249&cid=C039BDESELW) a page last week - https://restfulapi.net/rest-api-design-tutorial-with-example/

I think this API should be `{roleId}/permission/{permissionId}`

i.e: to associate a role and permission:

```
PUT
https://localhost:8888/api/v1/role/93601d0c-566f-4a79-af81-572d4d5ed3e4/permission/4490f8dc-1254-11ed-861d-0242ac120002
```

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/role/RoleEndpoint.kt [hunk 2]
```diff
@@ -47,22 +47,22 @@ interface RoleEndpoint : RpcOps {
...
-        @HttpRpcRequestBodyParameter(description = "Identifier for an existing permission")
+        @HttpRpcPathParameter(description = "Identifier for an existing permission")
         permissionId: String
     ): RoleResponseType
 
     /**
      * Removes Association between a role and a permission
      */
-    @HttpRpcPUT(path = "removePermission", description = "Removes a permission from a role")
+    @HttpRpcPUT(path = "removePermission/{roleId}/{permissionId}", description = "Removes a permission from a role")
```
Comment (sprintengine) [inline libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/role/RoleEndpoint.kt]: I think this should be a DELETE with same path as above:

`{roleId}/permission/{permissionId}`

This deletes a rrole-permission association. What about deleting a permission from the system? I think that'll look like:

`DELETE .../api/v1/permission/{permissionId}`

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/user/UserEndpoint.kt [hunk 3]
```diff
@@ -43,29 +43,29 @@ interface UserEndpoint : RpcOps {
     /**
      * Assign a Role to a User in the RBAC permission system.
      */
-    @HttpRpcPUT(path = "addRole", description = "Assign a Role to a User")
+    @HttpRpcPUT(path = "addRole/{loginName}/{roleId}", description = "Assign a Role to a User")
```
Comment (sprintengine) [inline libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/user/UserEndpoint.kt]: I think this should be `{loginName}/role/{roleId}`

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/user/UserEndpoint.kt [hunk 4]
```diff
@@ -43,29 +43,29 @@ interface UserEndpoint : RpcOps {
...
-        @HttpRpcRequestBodyParameter(description = "Id of the role to associate with this user")
+        @HttpRpcPathParameter(description = "Id of the role to associate with this user")
         roleId: String
     ): UserResponseType
 
     /**
      * Un-assign a Role from a User in the RBAC permission system.
      */
-    @HttpRpcPUT(path = "removeRole", description = "Un-assign a role from a user")
+    @HttpRpcPUT(path = "removeRole/{loginName}/{roleId}", description = "Un-assign a role from a user")
```
Comment (sprintengine) [inline libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/user/UserEndpoint.kt]: I think this should be DELETE `{loginName}/role/{roleId}`

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/user/UserEndpoint.kt [hunk 5]
```diff
@@ -43,29 +43,29 @@ interface UserEndpoint : RpcOps {
...
-        @HttpRpcRequestBodyParameter(description = "Id of the role to un-assign from this user")
+        @HttpRpcPathParameter(description = "Id of the role to un-assign from this user")
         roleId: String
     ): UserResponseType
 
     /**
      * Get a summary of a user's permissions.
      */
-    @HttpRpcGET(path = "permissionSummary/{loginName}", description = "Get a summary of a User's permissions")
+    @HttpRpcGET(path = "{loginName}/permissionSummary", description = "Get a summary of a User's permissions")
```
Comment (sprintengine) [inline libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/user/UserEndpoint.kt:69]: I think this is OK

File: libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/role/RoleEndpoint.kt (hunk 1 above)
Comment (sprintengine) [reply libs/permissions/permission-endpoint/src/main/kotlin/net/corda/libs/permissions/endpoints/v1/role/RoleEndpoint.kt]: That page also suggests renaming "role" to "roles", thats another separate thing to think about.

**Review by conal-smith — APPROVED** [review]
LGTM

**Pull request merged**